File: dotnet/Allmark/Utils/IsAlphaNumeric.cs
namespace Allmark;

public static partial class Utils
{
	public static bool IsAlpha(int code)
	{
		return (code > 64 && code < 91) || (code > 96 && code < 123);
	}

	public static bool IsNumeric(int code)
	{
		return code > 47 && code < 58;
	}

	public static bool IsAlphaNumeric(int code)
	{
		return IsAlpha(code) || IsNumeric(code);
	}
}

File: zig/src/utils/isAlphaNumeric.zig
const std = @import("std");

pub fn isAlpha(code: u8) bool {
    return (code > 64 and code < 91) or (code > 96 and code < 123);
}

pub fn isNumeric(code: u8) bool {
    return code > 47 and code < 58;
}

pub fn isAlphaNumeric(code: u8) bool {
    return isAlpha(code) or isNumeric(code);
}
